hydra-cardano-api/src/Hydra/Cardano/Api/AddressInEra.hs

```diff
@@ -76,10 +76,10 @@ toLedgerAddr = \case
     Ledger.Addr ntwrk creds stake
 
 -- | Convert a plutus 'Address' to an api 'AddressInEra'.
--- NOTE: Requires the 'Network' discriminator (Testnet or Mainnet) because
+-- NOTE: Requires the 'NetworkId' discriminator (Testnet or Mainnet) because
 -- Plutus addresses are stripped off it.
-fromPlutusAddress :: IsShelleyBasedEra era => Network -> Plutus.Address -> AddressInEra era
-fromPlutusAddress network plutusAddress =
+fromPlutusAddress :: IsShelleyBasedEra era => NetworkId -> Plutus.Address -> AddressInEra era
+fromPlutusAddress networkId plutusAddress =
```

Comment:
We used `Network` deliberately here. It's a smaller type than `NetworkId` and we only need to the discriminator to do (re-)create addresses correctly.

Must: Keep `Network` here and potentially update call sites.